internal/pkg/service/stream/migrate/source/source.go

```diff
@@ -0,0 +1,190 @@
+	api "github.com/keboola/keboola-as-code/internal/pkg/service/stream/api/gen/stream"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition/key"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+const (
+	createSourcePath    = "/v1/branches/default/sources"
+	receiversBufferPath = "/v1/receivers"
+	bufferPrefix        = "buffer"
+	streamPrefix        = "stream"
+)
+
+type Receivers struct {
+	Receivers []Receiver `json:"receivers"`
+}
+type Template struct {
+	Language string `json:"language"`
+	Content  string `json:"content"`
+}
+type Columns struct {
+	PrimaryKey bool     `json:"primaryKey"`
+	Type       string   `json:"type"`
+	Name       string   `json:"name"`
+	Template   Template `json:"template,omitempty"`
+}
+type Mapping struct {
+	TableID     string    `json:"tableId"`
+	Incremental bool      `json:"incremental"`
+	Columns     []Columns `json:"columns"`
+}
+type Conditions struct {
+	Count int    `json:"count"`
+	Size  string `json:"size"`
+	Time  string `json:"time"`
+}
+type Exports struct {
+	ID         string     `json:"id"`
+	ReceiverID string     `json:"receiverId"`
+	Name       string     `json:"name"`
+	Mapping    Mapping    `json:"mapping"`
+	Conditions Conditions `json:"conditions"`
+}
+type Receiver struct {
+	ID          string    `json:"id"`
+	URL         string    `json:"url"`
+	Name        string    `json:"name"`
+	Description string    `json:"description"`
+	Exports     []Exports `json:"exports"`
+}
+
+type RequestConfig struct {
+	method string
+	host   string
+	token  string
+	path   string
+	body   io.Reader
+	logger log.Logger
+}
+
+func New(method string, host string, token string, path string, body io.Reader, l log.Logger) RequestConfig {
+	return RequestConfig{
+		method: method,
+		host:   host,
+		token:  token,
+		body:   body,
+		path:   path,
+		logger: l,
+	}
+}
+
+func FetchBufferReceivers(ctx context.Context, host string, token string, logger log.Logger) (*Receivers, error) {
+	return fetchDataFromBuffer(ctx, New(
+		"GET",
+		"https://"+host,
+		token,
+		receiversBufferPath,
+		nil,
+		logger,
+	))
+}
+
+func fetchDataFromBuffer(ctx context.Context, reqConfig RequestConfig) (*Receivers, error) {
+	resp, err := newHTTPRequest(ctx, reqConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.Errorf(`unexpected status code %d from %s`, resp.StatusCode, reqConfig.host)
+	}
+
+	var result *Receivers
+
+	err = json.NewDecoder(resp.Body).Decode(&result)
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+func (r *Receiver) CreateSource(ctx context.Context, token string, host string, wg *sync.WaitGroup, l log.Logger) error {
+	defer wg.Done()
+
+	// Set a payload to create source
+	body, err := r.createSourcePayload()
+	if err != nil {
+		return err
+	}
+
+	streamHost := strings.Replace(host, bufferPrefix, streamPrefix, 1)
```

Comment:
This looks weird, could we get some example how `bufferPrefix` or `streamPrefix` looks like? Or maybe create `substituteHost` function so it is more convinient that the `Replace` has to be used when communicating.
Something like
```
func substituteHost(host, bufferPrefix, streamPrefix string) string {
  return "https://" + strings.Replace(host, bufferPrefix, streamPrefix, 1)
}
```

Reply:
[475667f](https://github.com/keboola/keboola-as-code/pull/1743/commits/475667ff4e3ce6df6dc3190038050551733e2e19)